lib/Request.js

```diff
@@ -310,3 +313,89 @@ Request.prototype.propertyError = function(property, errorCode, httpStatusCode,
         });
     }.bind(this));
 };
+
+Request.prototype.getStream = function() {
+    if (null === this._stream) {
+        this._stream = new Busboy({
+            headers: this._connection.raw.req.headers
+        });
+        this._connection.raw.req.pipe(this._stream);
+    }
+    return this._stream;
+}
+
+Request.prototype.getUpload = function(field) {
+    return new Promise(function(resolve, reject) {
+        var stream = this.getStream();
+        var found = false;
+
+        stream.on('file', function(uploadedFieldName, file, filename, encoding, mimetype) {
+            if (field != uploadedFieldName) {
+                // Got an upload, but not for the field name we are looking for.
+                // Skip this upload, maybe the next one will be under the desired field name.
+                return;
+            }
+
+            found = true;
```

Comment:
Yes, this works, and there's a unit test that validates the expected behavior:

    Request
      binary uploads
        getUpload()
          ✓ resolves with first upload by specified name if multiple uploads are sent with the same name 

Also, since promises can only ever be resolved once, calling resolve a second time for the second upload has no affect.

But that gives me an idea -- to improve this, we can stop listening to the `file` event once we find the matched upload. This will prevent unnecessary data chunking for the following uploads that match the given name.